eth_utils/hexidecimal.py

```diff
@@ -31,11 +30,15 @@ def decode_hex(value):
 def encode_hex(value):
     if not is_string(value):
         raise TypeError('Value must be an instance of str or unicode')
-    return add_0x_prefix(codecs.encode(value, 'hex'))
+    return add_0x_prefix(to_text(codecs.encode(value, 'hex')))
 
 
 def is_0x_prefixed(value):
-    return is_prefixed(value, '0x') or is_prefixed(value, '0X')
+    if not is_text(value):
+        raise TypeError(
+            "is_0x_prefixed requires text typed arguments. Got: {0}".format(type(value))
```

Comment:
My favorite way to show the value that triggered the error is a `repr(value)`, so that the caller can immediately see both the value and type.

Just sharing, it's up to you.

Reply:
makes sense to me, thanks